### ISSUE TITLE
---multi# magic URL to open two shared projects

### ISSUE THREAD

**Issue by Vegz78** [microsoft/pxt-microbit#5397]
**Is your feature request related to a problem? Please describe.**
When using the radio function in a classroom setting with the simulator, it is intuitive to use the https://makecode.microbit.org/---multi# editors and, for example, have the sender program in the left editor window and the receiver program in the right.

**Describe the solution you'd like**
I'd like to be able to make a link that opens 2 shared projects, the first in the left window and the second in the right window, maybe something like this:
https://makecode.microbit.org/---multi#_YTbKYAAimAxz&_LxghRpHJaKq8

**Describe alternatives you've considered**
I have found no alternatives.

**Additional context**
The feature has been discussed on the forum:
https://forum.makecode.com/t/editor-magic-url-flags/10743/5?u=vegz78

**Comment by bsiever** [comment]
I was just thinking about this too.  It could really streamline sharing some radio-based projects.

**Comment by Vegz78** [comment]
Yeah, @bsiever! -Especially for radio-based projects, this would be a great way to share them in an educational setting etc.!!! (@kiki-lee?)

I managed to open a project directly in the left split screen using the `---multi` URL magic flag, but haven't found yet how to also simultaneously open another project on the right side:
https://makecode.microbit.org/---multi#pub:_ismhAmAyA93c

Has anyone in or outside the makecode team found out yet?

**Comment by martinwork** [comment]
I notice the hot or cold sample does it with tutorials. 
https://makecode.microbit.org/projects/hot-or-cold-multi#:~:text=Start%20both%20tutorials%20side%2Dby%2Dside

I don't know why this link doesn't seem to work from here
https://makecode.microbit.org/---multi#pub:_YTbKYAAimAxz:|:pub:_LxghRpHJaKq8

But it works if pasted into the address bar, or from HTML
[makecode-multi.zip](https://github.com/user-attachments/files/17534933/makecode-multi.zip)
```
<!DOCTYPE html>
<script>
	async function exampleClick() {
	  try {
		window.open( "https://makecode.microbit.org/---multi#pub:_YTbKYAAimAxz:|:pub:_LxghRpHJaKq8", '_blank').focus();
	  } catch (error) {
		console.log(error);
	  }
	}
</script>

<html>
    <body>

        <h1>makecode--multi</h1>

        <p>Link</p>
		
		<a href="https://makecode.microbit.org/---multi#pub:_YTbKYAAimAxz:|:pub:_LxghRpHJaKq8">https://makecode.microbit.org/---multi#pub:_YTbKYAAimAxz:|:pub:_LxghRpHJaKq8</a>
		
        <p>Button</p>
		
		<button onclick="exampleClick()">Example</button>
				
    </body>
</html>
```

**Comment by martinwork** [comment]
The link doesn't work from here because the invisible link has %7C instead of |
https://makecode.microbit.org/---multi#pub:_YTbKYAAimAxz:|:pub:_LxghRpHJaKq8

`https://makecode.microbit.org/---multi#pub:_YTbKYAAimAxz:%7C:pub:_LxghRpHJaKq8`

**Comment by Vegz78** [comment]
You are a special kind of genius destined for greatness, mr. @martinwork!

Great catch and thank you very much! Been looking for a solution to this for ages!

Seems like the ASCII vertical bar is an unsafe character which is force converted to %7C in URLs, at least here on GitHub, but works under the makecode domains and webapps, but not the reencoded %7C. Interesting!

How to make links with vertical bars work in general, then?

**Comment by martinwork** [comment]
Whoever created it is the genius... I just searched the source for multi"!

I found another use here
https://github.com/KitronikLtd/pxt-kitronik-stopbit/blob/master/RadioCrossingLeft.md?plain=1#L10

I think this might be the only place that checks for "|" - actually ":|:".
https://github.com/microsoft/pxt/blob/master/webapp/public/multi.html#L74

Could it also try ":%7C:"?

**Comment by Vegz78** [comment]
> Could it also try ":%7C:"?

Good idea!

@abchatra: Would it be possible to please assign someone on your team on this issue to do the quick change so that the various MakeCode editors responds the same way to `:%7C:` as to `:|:`, and it becomes possible to publish links that open two shared and interconnected MakeCode projects directly in the `---multi` split screen mode also on many websites that forcibly reencodes URLs with the `|` (verrtical bar) to `%7C`?
